src/pages/ReimbursementAccount/BankAccountStep.js

```diff
@@ -158,7 +170,9 @@ class BankAccountStep extends React.Component {
         // Disable bank account fields once they've been added in db so they can't be changed
         const isFromPlaid = this.props.achData.setupType === CONST.BANK_ACCOUNT.SETUP_TYPE.PLAID;
         const shouldDisableInputs = Boolean(this.props.achData.bankAccountID) || isFromPlaid;
-        const subStep = this.props.achData.subStep;
+        const shouldReinitializePlaidLink = this.props.existingPlaidLinkToken && this.props.receivedRedirectURI;
+        const subStep = shouldReinitializePlaidLink ? CONST.BANK_ACCOUNT.SETUP_TYPE.PLAID : this.props.achData.subStep;
```

Comment:
Might be helpful to explain briefly why this is necessary? It's not immediately clear to me why we should not use `achData.subStep` when we are returning to the flow. Is there a situation where the value is something other than `plaid`?

Reply:
Yeah, so I'd need to investigate more why this is the case but `subStep` coming back from the OAuth flow is null. Perhaps because a bank account hasn't been selected yet we haven't updated the `reimbursementAccount` key.

<img width="712" alt="Screen Shot 2021-11-29 at 3 54 46 PM" src="https://user-images.githubusercontent.com/24466196/143943484-3bd52f00-ea45-4b81-9b33-91d9ac25d02c.png">

Reply:
^at this moment we need to reinitialize the PlaidLink

Reply:
Ok cool. We don't _need_ to worry about it right now. But feels like the problem could possibly be solved some better way. Perhaps by ensuring that the `subStep` is set to `plaid` after selected - but unsure if this is possible and haven't looked too deeply at this.

Reply:
> Perhaps by ensuring that the subStep is set to plaid after selected 

Yeah, was thinking the same thing